packages/adapters/ethers5/src/client.ts

```diff
@@ -361,12 +361,30 @@ export class Ethers5Adapter extends AdapterBlueprint {
         method: 'eth_chainId'
       })
 
+      if (requestChainId !== chainId) {
+        const caipNetwork = this.caipNetworks?.find(n => n.id === chainId)
+
+        if (!caipNetwork) {
+          throw new Error('Ethers5Adapter:connect - could not find the caipNetwork to switch')
+        }
+
+        try {
+          await this.switchNetwork({
+            caipNetwork,
+            provider: selectedProvider,
+            providerType: type as ConnectorType
+          })
+        } catch (error) {
+          throw new Error('Ethers5Adapter:connect - Switch network failed')
+        }
+      }
```

Comment:
can't the connection be abstracted  to `EthersMethoids` and pulled from there from ethers and ethers5?